discovery-infra/test_infra/kubeapi_utils.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import functools
+
+import waiting
+
+from ipaddress import IPv4Interface, IPv6Interface
+
+from test_infra import utils
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_ip_for_single_node(cluster_deployment, is_ipv4, timeout=300):
+    agents = cluster_deployment.list_agents()
+    assert len(agents) == 1
+    agent = agents[0]
+
+    def get_bmc_address():
+        interfaces = agent.status().get('inventory', {}).get('interfaces')
+        if not interfaces:
+            return
+        ip_addresses = interfaces[0].get(
+            'ipV4Addresses' if is_ipv4 else 'ipV6Addresses'
+        )
+        if not ip_addresses:
+            return
+        interface_class = IPv4Interface if is_ipv4 else IPv6Interface
+        return str(interface_class(ip_addresses[0]).ip)
+
+    return waiting.wait(
+        get_bmc_address,
+        sleep_seconds=0.5,
+        timeout_seconds=timeout,
+        waiting_for=f'single node ip of agent {agent.ref}',
+    )
+
+
+def set_agents_hostnames(
+        cluster_deployment,
+        is_ipv4,
+        static_network_mode,
+        tf,
+        nodes_number,
+):
+    if is_ipv4 and not static_network_mode:
+        return
+
+    logger.info('Updating agents hostnames')
+    nodes_details = get_nodes_details(
+        cluster_name=cluster_deployment.ref.name,
+        namespace=cluster_deployment.ref.namespace,
+        tf=tf,
+    )
+    for agent in cluster_deployment.wait_for_agents(nodes_number):
+        set_agent_hostname(agent, nodes_details)
+
+
+def get_nodes_details(cluster_name, namespace, tf):
+    tf_folder = utils.get_tf_folder(cluster_name, namespace)
+    tf_vars = utils.get_tfvars(tf_folder)
+    networks_names = (
+        tf_vars['libvirt_network_name'],
+        tf_vars['libvirt_secondary_network_name'],
+    )
+    return utils.get_libvirt_nodes_from_tf_state(
+        network_names=networks_names,
+        tf_state=tf.get_state(),
+    )
+
+
+def set_agent_hostname(agent, nodes_details):
+    assert len(nodes_details) >= 1
+    hostname = waiting.wait(
+        functools.partial(get_hostname_for_agent, nodes_details, agent),
+        timeout_seconds=60,
+        sleep_seconds=1,
+        waiting_for=f'agent={agent.ref} to find a hostname',
+    )
+    logger.info('patching agent hostname=%s', hostname)
+    agent.patch(hostname=hostname)
+
+
+def get_hostname_for_agent(nodes_details, agent):
```

Comment:
It bugs me that the order of the parameters is not the same as above function

Reply:
bless you